>Ryan Miyahara
>UCID:804585999
>The change made was a substitution of a subtraction statement for a compare
>statement. This was done to fix a bug pertaining an overflow error if
>b.tv_nsec was much larger than a.tv_nsec.
--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2018-02-08 18:14:02.384626000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_nsec > b.tv_nsec)
+    return 1;
+  else if (a.tv_nsec == b.tv_sec)
+    return 0;
+  else
+    return -1;
 }

 # if ! HAVE_DECL_NANOSLEEP
